src/types.d.ts

```diff
@@ -19,7 +30,82 @@ interface StyleElement {
 	Outline?: number;
 	Shadow?: number;
 	Alignment?: number;
-}
-
-export function convertToASS(time: string, options: SyllabesConfig): string
-
+	ScaleX?: number;
+	ScaleY?: number;
+	Spacing?: number;
+	Angle?: number;
+	BorderStyle?: number;
+	MarginL?: number;
+	MarginR?: number;
+	MarginV?: number;
+	AllCaps?: boolean; // Virtual attribute, used during conversion, not part of ass file
+  }
+  interface IStyleKV {
+	key: "Style";
+	value: IStyle;
+  }
+  interface IStyles {
+	section: string;
+	body: Array<{ key: "Format"; value: Array<keyof IStyle> } | IStyleKV>;
+  }
+  interface IDialogue {
+	Layer: string;
+	Start: string;
+	End: string;
+	Style: string;
+	Name: string;
+	MarginL: string;
+	MarginR: string;
+	MarginV: string;
+	Effect: string;
+	Text: string;
+  }
+  interface IDialogueKV<T> {
+	key: T;
+	value: IDialogue;
+  }
+  interface IFormatKV {
+	key: "Format";
+	value: Array<keyof IDialogue>;
+  }
+  
+  // ? same as sentence ?
```

Comment:
I don't understand this comment ?

Reply:
Good question. It was part of Ian's big type-ification commit and I must not have noticed it. It doesn't mean anything to me so I'll remove it.